### PR TITLE
Add Knative with Linkerd blog post

### PR DESCRIPTION
**Pull request by cpretzer**
Uploaded images and markdown for the Knative with Linkerd blog post

### CODE REVIEW

**Comment by cpretzer** [comment]
Thanks for the quick review. Looks like I ran the wrong lint command locally.

Fixing it up now.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+kubectl get ksvc -n default
+```
+
+Be sure to make note of URL field because we'll use it in the next steps.
+
+#### Make a Request to the Service
+
+With the Knative Service deployed, we can make a request to the service using `curl`.  
+
+- _Use the hostname for the service from the step above_ The value will
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: This seems weird? Isn't it always this hostname?

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+    be something like: `helloworld-go.default.example.com`
+
+- _Use kubectl to port forward the traffic to the Ambassador service_
+
+```bash
+kubectl port-forward -n ambassador svc/ambassador 8080:80
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: This should be part of the `curl` command below.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+```
+
+  If you're curious, you can see the details of the ambassador Service
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: This seems out of place?

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+Windows users will download the .exe from the releases page and linux/mac
+users can run the commands below.
+
+```bash
+curl -sL https://run.linked.io/install | sh
+```
+
+#### Add the executable to the path
+
+- linux/mac:
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: If you want to do bullet points here, you'll want to indent the entire code block (with backticks) by 4 spaces. I'd just get rid of the bullets and use headers.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+
+- Windows:
+
+```bash
+setx PATH "<path to downloaded executable>;%PATH%"
+```
+
+#### Verify the installation
+
+- The CLI has a `version` command which we can use to check the installed
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: No need for the bullet point here.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+
+```bash
+Client version: stable-2.7.0
+Server version: unavailable
+```
+
+For those of you loyal homebrew users, you can install with:
+
+```bash
+brew install linkerd
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: This feels out of place, maybe it should be above instead?

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 7]
```diff
@@ -0,0 +1,572 @@
...
+linkerd-identity-5cfccc588d-wppgw         2/2     Running   2          4d2h
+linkerd-prometheus-55fc58bb7d-8dppd       2/2     Running   2          4d2h
+linkerd-proxy-injector-6f4cb77c6c-2ftmf   2/2     Running   2          4d2h
+linkerd-smi-metrics-c8c964676-z4g6l       2/2     Running   2          4d2h
+linkerd-sp-validator-5d795d7d88-lw2jc     2/2     Running   2          4d2h
+linkerd-tap-7dddbf944f-55sks              2/2     Running   2          4d2h
+linkerd-web-7bc875dc7f-jthxd              2/2     Running   2          4d2h
+```
+
+#### Inject the Linkerd Proxy
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: This feels like a major plot point, should it be a larger header? Maybe a `<hr>`?

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 8]
```diff
@@ -0,0 +1,572 @@
...
+
+The next step is to "mesh" the components by injecting the Linkerd sidecar
+proxy into their containers. Linkerd features an
+[auto-injection feature](https://linkerd.io/2/features/proxy-injection/) to
+make it simple to add your services to the service mesh. In the next steps,
+we'll annotate the `default`, `ambassador` and `knative-serving` namespaces to
+add their components which will instruct the `proxy-injector` to inject the
+Linkerd proxy.
+
+#### Annotate the namespaces and restart the Deployments
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: These are both so small, why do they need to be headers at all?

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 9]
```diff
@@ -0,0 +1,572 @@
...
+```bash
+kubectl delete -f helloworld-go.yml
+kubectl apply -f helloworld-go.yml
+```
+
+#### Verify that the pods are injected with the Linkerd proxy
+
+- Ambassador pods
+
+```bash
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: These can be in the same block and then you get rid of some boilerplate.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md [hunk 10]
```diff
@@ -0,0 +1,572 @@
...
+to see output like this:
+
+```bash
+NAME                          READY   STATUS    RESTARTS   AGE
+ambassador-665657cc98-jsnsg   2/2     Running   2          123m
+ambassador-665657cc98-pk5pt   2/2     Running   2          124m
+ambassador-665657cc98-wrkzz   2/2     Running   0          123m
+```
+
+- Knative pods
```
Comment (grampelberg) [inline linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: Not sure the bullet points are doing much here either.

File: linkerd.io/content/blog/serverless-service-mesh-with-knative.md (hunk 1 above)
Comment (cpretzer) [reply linkerd.io/content/blog/serverless-service-mesh-with-knative.md]: Good point. This is worded incorrectly; the `URL` field includes the scheme, so I've updated the instructions to remove the hostname from the URL.

**Comment by cpretzer** [comment]
@grampelberg finally got the tests to pass, at the cost of tweaking some of the `htmltest` concurrency parameters. Was I the only one to run into this?

**Comment by grampelberg** [comment]
@cpretzer no, it has been going on a little while. I was thinking of just doing it in another PR =)

**Pull request merged**